src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -476,6 +496,33 @@ private fun resolveAstNode(
             if (type.isAbstract()) {
                 type = defaultConcretizationOf(type)
             }
+
+            if (node.accessMode != null && node.addressSpace == null) {
+                throw IllegalArgumentException(
+                    "If an access mode is specified for a variable an address space must also be specified.",
+                )
+            } else if (node.addressSpace in
+                setOf(
```

Comment:
Given that this is the complement of the set just containing AddressSpace.FUNCTION, it would be clearer to compare node.addressSpace with AddressSpace.FUNCTION and complain if they don't match.